models/invoice_number.go

```diff
@@ -0,0 +1,43 @@
+package models
+
+import (
+	"log"
+	"strings"
+
+	"github.com/jinzhu/gorm"
+)
+
+type InvoiceNumber struct {
+	ID     string
+	Number int64
+}
+
+// TableName returns the database table name for the LineItem model.
+func (InvoiceNumber) TableName() string {
+	return tableName("invoice_numbers")
+}
+
+// NextInvoiceNumber updates and returns the next invoice number for the instance
+func NextInvoiceNumber(tx *gorm.DB, instanceID string) (int64, error) {
+	number := InvoiceNumber{}
+	if instanceID == "" {
+		instanceID = "global-instance"
+	}
+
+	if result := tx.Where(InvoiceNumber{ID: instanceID}).Attrs(InvoiceNumber{Number: 0}).FirstOrCreate(&number); result.Error != nil {
+		return 0, result.Error
+	}
+
+	if result := tx.Raw("select number from \""+number.TableName()+"\" where id = ? for update", instanceID).Scan(&number); result.Error != nil {
```

Comment:
There is a way to get a quoted table name from `gorm`. See https://github.com/netlify/gocommerce/blob/753bfc577552c7044c375a2a1af8db81846ea920/api/params.go#L55 for an example.